ta/pkcs11/src/pkcs11_attributes.c

```diff
@@ -707,13 +707,13 @@ create_attributes_from_template(struct obj_attrs **out, void *template,
 	}
 #endif
 
-	rc = sanitize_client_object(&temp, template, template_size);
-	if (rc)
-		goto out;
-
-	/* If class/type not defined, match from mechanism */
-	if (get_class(temp) == PKCS11_UNDEFINED_ID &&
-	    get_key_type(temp) == PKCS11_UNDEFINED_ID) {
+	/*
+	 * For PKCS11_FUNCTION_GENERATE, find the class and type
+	 * based on the mechanism. These will be passed as hint
+	 * sanitize_client_object() and added in temp if not
+	 * already present
+	 */
+	if (function == PKCS11_FUNCTION_GENERATE) {
 		switch (mecha) {
```

Comment:
Can't add comment above in function signature ;)

But please remove '__unused.

Reply:
Sure, I will remove that.  I have been planning to do so whenever I see that but somehow it gets skipped.

Reply:
Done in latest PR